[FrameworkBundle] Respect debug mode when warm up annotations

Strate opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |


Propagate current debug mode to the annotations reader. Without thi, warmup is useless with debug mode, because timetsamps are not written to cache.

Strate committed:
[FrameworkBundle] Respect debug mode when warm up annotations

iltar commented on src/Symfony/Bundle/FrameworkBundle/CacheWarmer/AnnotationsCacheWarmer.php:
```diff
@@ -28,17 +28,20 @@ class AnnotationsCacheWarmer extends AbstractPhpFileCacheWarmer
+     * @param bool                   $debug            Run in debug mode
      */
-    public function __construct(Reader $annotationReader, $phpArrayFile, CacheItemPoolInterface $fallbackPool, $excludeRegexp = null)
+    public function __construct(Reader $annotationReader, $phpArrayFile, CacheItemPoolInterface $fallbackPool, $excludeRegexp = null, $debug)
```
You've added a non-optional argument, which _should_ have broken tests. If there are none, can you add one to prevent future regressions? As this argument is non-optional, it would also be a BC break. Can you make this argument optional with a default value to mimic the previous behavior?

Strate replied:
All done.
Added test for cache warmer for both cases - debug and no debug
Made argument optional

Strate committed:
[FrameworkBundle] Respect debug mode when warm up annotations

Make new argument for AnnotationsCacheWarmer::__construct optional to prevent BC break

Add tests

Strate committed:
[FrameworkBundle] Respect debug mode when warm up annotations

Code style

Strate committed:
[FrameworkBundle] Respect debug mode when warm up annotations

Extracted reader mock to method, made test compatible with old php-unit

Strate commented:
Just pushed fix for old phpunit, but seems it was useless as changes already aproved (travis ran on old code)

nicolas-grekas commented:
Still fails on PHP 5.5, missing "getMock()"

Strate committed:
[FrameworkBundle] Respect debug mode when warm up annotations

tests compatibility with php 5.5

Strate commented:
That's all, travis is happy now.

fabpot commented:
Thank you @Strate.
